diff --git a/debian/control b/debian/control
index b2c9f34..643ce04 100644
--- a/debian/control
+++ b/debian/control
@@ -17,7 +17,6 @@ Build-Depends: debhelper-compat (= 12), dh-python, python3:any, dracut-core, qui
                gettext,
                gperf,
                git,
-               gnu-efi [amd64 i386 arm64 armhf],
                libcap-dev (>= 1:2.24-9~),
                libpam0g-dev,
                libapparmor-dev (>= 2.9.0-3+exp2) <!stage1>,
@@ -41,7 +40,7 @@ Build-Depends: debhelper-compat (= 12), dh-python, python3:any, dracut-core, qui
                libkmod-dev (>= 15),
                libblkid-dev (>= 2.24),
                libmount-dev (>= 2.30),
-               libseccomp-dev (>= 2.3.1) [amd64 arm64 armel armhf i386 mips mipsel mips64 mips64el x32 powerpc ppc64 ppc64el s390x],
+               libseccomp-dev (>= 2.3.1) [amd64 arm64 armel armhf i386 mips mipsel mips64 mips64el x32 powerpc ppc64 ppc64el s390x riscv64],
                libdw-dev (>= 0.158) <!stage1>,
                libpolkit-gobject-1-dev <!stage1>,
                linux-base <!nocheck>,
@@ -65,8 +64,8 @@ Standards-Version: 4.4.1
 Homepage: https://launchpad.net/ubuntu-core-initramfs
 
 Package: ubuntu-core-initramfs
-Architecture: amd64 arm64 armhf
+Architecture: amd64 arm64 armhf riscv64
Depends: ${python3:Depends}, ${misc:Depends}, binutils, dracut-core (>= 048-1), lz4, linux-firmware
 Description: standard embedded initrd
  Standard embedded initrd implementation to be used with Ubuntu Core
  systems. Currently targetting creating BLS Type2 like binaries.
diff --git a/debian/install b/debian/install
index 54b4fc6..2cd3e89 100644
--- a/debian/install
+++ b/debian/install
@@ -2,6 +2,5 @@ bin/ubuntu-core-initramfs usr/bin
 postinst.d etc/kernel/
 snakeoil/* usr/lib/ubuntu-core-initramfs/snakeoil/
 debian/tmp/* usr/lib/ubuntu-core-initramfs/main
-debian/tmp/usr/lib/systemd/boot/efi/linux*.efi.stub usr/lib/ubuntu-core-initramfs/efi/
 debian/sbat.txt usr/lib/ubuntu-core-initramfs/efi/
 features/* usr/lib/ubuntu-core-initramfs/
diff --git a/debian/rules b/debian/rules
index 35c6f74..b83c67c 100755
--- a/debian/rules
+++ b/debian/rules
@@ -34,7 +34,7 @@ override_dh_auto_configure:
       -Dhibernate=false \
       -Dldconfig=false \
       -Dresolve=false \
-      -Defi=true \
+      -Defi=false \
       -Dtpm=true \
       -Dtpm-pcrindex=12 \
       -Denvironment-d=false \
@@ -110,15 +110,12 @@ override_dh_auto_configure:
       -Dpcre2=true \
       -Dglib=false \
       -Ddbus=false \
-      -Dgnu-efi=true \
+      -Dgnu-efi=false \
       -Duserdb=false \
       -Dbashcompletiondir=no \
       -Dzshcompletiondir=no \
       -Dinstall-tests=false
 
-override_dh_auto_test:
-	LD_LIBRARY_PATH= LD_PRELOAD= dh_auto_test --buildsystem=meson+ninja --sourcedirectory=vendor/systemd || :
-
 override_dh_auto_clean:
 	dh_auto_clean --buildsystem=meson+ninja --sourcedirectory=vendor/systemd
 	sed -i '/"minix"/s/^\/*//' vendor/systemd/src/test/test-path-util.c
@@ -231,7 +228,6 @@ ifeq ($(DEB_HOST_ARCH_BITS),64)
 	GOPATH=$(CURDIR)/vendor/gopath/ HOME=$(CURDIR)/debian CGO_ENABLED=1 go test -buildmode=pie -tags nomanagers github.com/snapcore/snapd/cmd/snap-bootstrap || :
 endif
 endif
-	rm -rf debian/ubuntu-core-initramfs/usr/lib/ubuntu-core-initramfs/main/usr/lib/systemd/boot
 ifeq ($(DEB_HOST_ARCH),amd64)
 	mkdir -p debian/ubuntu-core-initramfs/usr/lib/ubuntu-core-initramfs/early/
 	debian/generate-x86-microcode debian/ubuntu-core-initramfs/usr/lib/ubuntu-core-initramfs/early/microcode.cpio
@@ -256,3 +252,5 @@ override_dh_shlibdeps:
 	dh_shlibdeps -Xusr/lib/ubuntu-core-initramfs/main
 override_dh_strip:
 	dh_strip -Xsnap-bootstrap
+override_dh_auto_test:
+	dh_auto_test --no-parallel
